File: JavaScript/Math-Array-String/78.js
// 771. Jewels and Stones

/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function(jewels, stones) {
    let jewelSet = new Set(jewels); // Convert jewels to a Set for quick lookup
    return [...stones].filter(stone => jewelSet.has(stone)).length; // Count occurrences
};

// Example test cases
console.log(numJewelsInStones("aA", "aAAbbbb")); // Output: 3
console.log(numJewelsInStones("z", "ZZ"));       // Output: 0

File: Python/Math-Array-String/78.py
# 771. Jewels and Stones

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewel_set = set(jewels)  # Convert jewels to a set for O(1) lookups
        return sum(stone in jewel_set for stone in stones)  # Count how many stones are in the jewel set
    
sol = Solution()
output = sol.numJewelsInStones("aA", "aAAbbbb")
print(output)  # Output: 3